### DISCUSSION TITLE
How to customize a new workflow

### DISCUSSION THREAD

**Discussion by hadesfgh** [open-mmlab/mmediting#1344] [How-to] [answered]
In mmediting, we can only use train or val workflow. However, I have some specific jobs. Can you tell me how to customize a new workflow.

**Comment by zengyh1900** [comment]
Hi, 

In addition to training, testing, and evaluation, what kind of workflow do you want? Maybe inference?
We have provided some APIs to quickly run some demos, you can check them here: https://mmediting.readthedocs.io/en/1.x/user_guides/3_inference.html 

Hope it helps!

**Reply by hadesfgh** [reply]
I want to add a new workflow in the configuration file, such as workflow = [('train', 1), ('train_v2', 1)]. Specifically, train_v2 is a new kind train workflow with a new train_pipeline.  train_v2 pipeline and train pipeline train the network together.

**Comment by LeoXing1996** [comment, marked as answer]
Hi @hadesfgh , if you want to custom your workflow, I strongly recommend you to attempt [1.x version of MMEditing](https://github.com/open-mmlab/mmediting/tree/1.x). 
In MMEditing-1.x, we use a class named `Loop` to handle the workflow and you can define your own `Loop`.
* [`Loop` in MMEditing](https://github.com/open-mmlab/mmediting/tree/1.x/mmedit/engine/runner)
* [`Loop` in MMEngine](https://github.com/open-mmlab/mmengine/blob/main/mmengine/runner/loops.py) and [tutorial (only Chinese version available currently)](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/tutorials/runner.md).

**Reply by hadesfgh** [reply]
thank you for your relpy, i will check the [1.x version of MMEditing](https://github.com/open-mmlab/mmediting/tree/1.x) soon.
